docs/DeveloperGuide.md

```diff
@@ -1,38 +1,564 @@
+
+- `find-tx /s finished`: Lists down all the transactions that have been completed.
+- `find-tx /s unfinished`: Lists down all the transactions that are currently still ongoing.
+
+Given below is an example usage scenario and how the command mechanism behaves at each step.
+
+Step 1: The user types in the command in the command line. The CommandParser class checks if the command is valid through the createCommand() method, and either sends an exception, or send the input to the ViewTransactionsByStatus command to be processed.
+
+Step 2: The ViewTransactionsByStatus command checks if the delimiter ('s') is present in the user input through the getArgs() method. If not present, an exception will be thrown. The command also checks whether the input's final argument is 'finished' or 'unfinished' through the isValidArgument() method. An exception will also be thrown if the final argument does not match the required words.
+
+Step 3: Assuming the final argument is 'finish', the entire transactionList will be iterated through, and an array will store the transaction if the transaction.isFinished() status is true.
+
+Step 4: Once the transactionList has been completely iterated through, a message will be displayed to the user via Ui.viewcompletedTransactionsMessage() which lists down all the transactions that have their isFinished() status set to true.
+
+The following sequence diagram shows how the viewTransactionsByStatus operation works:
+
+![viewTransactionsByStatus](images/ViewTransactionByStatusSequence.png)
+
+#### 4.3.6. Update an transaction
+
+> This feature allows the user to update a specific item in the item list
+
+Given below is an example usage scenario and how the command mechanism behaves at each step.
+
+Step 1: The user types in the command in the command line. The CommandParser class checks if the command is valid through the createCommand() method, and either throws an exception, or forwards the input to `UpdateTransactionCommand` class to be processed.
+
+Step 2: `UpdateTransactionCommand::executeCommand` checks if the delimiters ('t', 'd') is present in the user input with the `getArgsViewItemCmd()` method. If any arguments are not present, an exception will be thrown.
+
+Step 3: If all commands are present, it then checks if arguments are valid with `UpdateTransactionCommand::areValidArgs` . An exception will also be thrown if the final argument does not satisfy the requirements (transaction not found or duration not valid).
+
+Step 4: If all arguments are valid, it then delegates to `TransactionList::updateTransactionDuration` which finds the transaction in the list by `transactionId`, which is unique and calls `Transaction::updateDuration`, which returns a Transaction with updated duration
+
+Step 5: `TransactionList` then updates itself with the latest item and returns
+
+Step 6: Lastly, information about the updated transaction is printed to the user
+
+The following sequence diagram models the operation:
```

Comment:
Would it be better to leave out the exception call details as they are repeated multiple times throughout the docs